mongoengine/context_managers.py

```diff
@@ -257,3 +258,13 @@ def set_write_concern(collection, write_concerns):
     combined_concerns = dict(collection.write_concern.document.items())
     combined_concerns.update(write_concerns)
     yield collection.with_options(write_concern=WriteConcern(**combined_concerns))
+
+
+@contextmanager
+def set_read_write_concern(collection, write_concerns, read_concern):
```

Comment:
Please add a test for this method, just asserting that the collection that is returned is configured properly